store/src/java/com/zimbra/cs/service/mail/SendShareNotification.java

```diff
@@ -758,13 +759,19 @@ private void processDBFilesSharedWithMe(MailItem reqItem, Action action, Operati
 
         Document docReqItem = (Document) reqItem;
         FileSharedWithMeRequest req = new FileSharedWithMeRequest();
+        String contentType = null;
+        // if file is in txt format convert to docx format
+        if (StringUtils.endsWith(fileName, ".txt") && docReqItem.getContentType().equals(MimeConstants.CT_TEXT_PLAIN)) {
```

Comment:
@gabhinavTI what happens if only txt file is shared, as the conversion of txt to docx happens from editing. Now if the txt is shared without editing will this valid to change to extension and content type?

Reply:
As per current code scenario, even if txt file is shared then we get a pop up that the file will be changed to .docx after editing
SO considering this the changes were in place.
If this change here is not required, then code in doc revision will take care of modifying the filename after editing

```
Convert .txt to .docx
This file will be converted to .docx format after editing.
```

Reply:
@dasiyogesh and @gabhinavTI let's have a quick call and discuss this PR.

Reply:
Discussed and checked with Mitesh, this change is not required
the file will be shared as .txt only
and once file is edited and renames as .docx , the updation happens accordingly